crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -793,6 +794,47 @@ impl Starknet {
         }
     }
 
+    pub fn update_gas(&mut self, gas_prices: GasUpdate) -> DevnetResult<GasUpdate> {
+        // BlockContext needs to be reinitialized
+        self.block_context = Starknet::init_block_context(
+            gas_prices.gas_price_wei,
+            gas_prices.gas_price_strk,
+            gas_prices.data_gas_price_wei,
+            gas_prices.data_gas_price_strk,
+            constants::ETH_ERC20_CONTRACT_ADDRESS,
+            constants::STRK_ERC20_CONTRACT_ADDRESS,
+            ChainId::from_str(self.block_context.chain_info().chain_id.0.as_str())
+                .expect("Chain id not supported"),
```

Comment:
Unacceptable expect. Better to have a converter in chain_id.rs

Reply:
refactored in 59d3925a2e8642d9717e08157de059fbb8a1b2d1